### PR TITLE
src/rgw: Fixing event name

### PR DESCRIPTION
**Pull request by TRYTOBE8TME**
Received as "Noncurrent" instead of "NonCurrent"

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by TRYTOBE8TME** [comment]
This is a suggested fix for: https://tracker.ceph.com/issues/59488

File: src/rgw/rgw_notify_event_type.cc [hunk 1]
```diff
@@ -30,8 +30,10 @@ namespace rgw::notify {
       return "s3:ObjectLifecycle:Expiration:*";
     case ObjectExpirationCurrent:
       return "s3:ObjectLifecycle:Expiration:Current";
+    case ObjectExpirationNonCurrent:
+      return "s3:ObjectLifecycle:Expiration:NonCurrent";
     case ObjectExpirationNoncurrent:
```
Comment (yuvalif) [inline src/rgw/rgw_notify_event_type.cc:35]: please remove this enum

File: src/rgw/rgw_notify_event_type.cc [hunk 2]
```diff
@@ -40,8 +42,10 @@ namespace rgw::notify {
       return "s3:ObjectLifecycle:Transition:*";
     case ObjectTransitionCurrent:
       return "s3:ObjectLifecycle:Transition:Current";
+    case ObjectTransitionNonCurrent:
+      return "s3:ObjectLifecycle:Transition:NonCurrent";
     case ObjectTransitionNoncurrent:
```
Comment (yuvalif) [inline src/rgw/rgw_notify_event_type.cc:47]: please remove this enum

**Review by yuvalif — APPROVED** [review]

**Comment by TRYTOBE8TME** [comment]
jenkins test window

**Comment by TRYTOBE8TME** [comment]
jenkins test windows

**Comment by TRYTOBE8TME** [comment]
Teuthology run: https://pulpito.ceph.com/kalpesh-mac-2024-05-23_09:09:47-rgw:notifications-wip-bug-59488-distro-default-smithi/

**Comment by yuvalif** [comment]
> Teuthology run: https://pulpito.ceph.com/kalpesh-mac-2024-05-23_09:09:47-rgw:notifications-wip-bug-59488-distro-default-smithi/

the LC failure is a known one. however, since this chnage the LC notification code, lets try to rerun. or wait for one fo these fixes:
https://github.com/ceph/ceph/pull/57638
https://github.com/ceph/ceph/pull/57632

**Comment by yuvalif** [comment]
@TRYTOBE8TME can you please rebase and rerun teuthology?
with the latest test fixes, tests seems to be passing more often.

**Comment by TRYTOBE8TME** [comment]
> @TRYTOBE8TME can you please rebase and rerun teuthology? with the latest test fixes, tests seems to be passing more often.

Here's the run:
https://pulpito.ceph.com/kalpesh-mac-2024-06-06_14:49:27-rgw:notifications-wip-bug-59488-distro-default-smithi/

**Comment by yuvalif** [comment]
> > @TRYTOBE8TME can you please rebase and rerun teuthology? with the latest test fixes, tests seems to be passing more often.
> 
> Here's the run: https://pulpito.ceph.com/kalpesh-mac-2024-06-06_14:49:27-rgw:notifications-wip-bug-59488-distro-default-smithi/

the kafka failure is a known crash:
```
Fatal glibc error: tpp.c:87 (__pthread_tpp_change_priority): assertion failed: previous_prio == -1 || (previous_prio >= fifo_min_prio && previous_prio <= fifo_max_prio)
```
the LC test failure is a known issue. LC test are passing locally, so issue is not introduced by the test.

**Pull request merged**